lib/std/hash_map.zig

```diff
@@ -1225,14 +1233,23 @@ pub fn HashMapUnmanaged(
         /// If there is an `Entry` with a matching key, it is deleted from
         /// the hash map, and this function returns true.  Otherwise this
         /// function returns false.
+        ///
+        /// TODO: answer the question in these doc comments, does this
+        /// increase the unused capacity by one?
         pub fn remove(self: *Self, key: K) bool {
             if (@sizeOf(Context) != 0)
                 @compileError("Cannot infer context " ++ @typeName(Context) ++ ", call removeContext instead.");
             return self.removeContext(key, undefined);
         }
+
+        /// TODO: answer the question in these doc comments, does this
+        /// increase the unused capacity by one?
         pub fn removeContext(self: *Self, key: K, ctx: Context) bool {
             return self.removeAdapted(key, ctx);
         }
+
+        /// TODO: answer the question in these doc comments, does this
+        /// increase the unused capacity by one?
```

Comment:
If you look at `removeByIndex`, which is what this calls, you'll see it decrements `size`, and increments `available`, meaning unused capacity is directly increased by one, yeah. Ditto for other functions that call this function or `removeByIndex` transitively.

Reply:
to solve that followup issue it will need the doc comments added and also unit test coverage to ensure it does not regress